app/controllers/lettings_logs_controller.rb

```diff
@@ -96,4 +115,13 @@ def find_resource
   def post_create_redirect_url(log)
     lettings_log_url(log)
   end
+
+  def mark_logs_resolved
+    if @log&.unresolved == true && @log.location.present? && @log.scheme.present? && @log.update(unresolved: false)
+      notice_message = "You’ve updated all the fields affected by the scheme change."
+      unresolved_logs_count_for_user = current_user.lettings_logs.where(unresolved: true, created_by: current_user).count
+      notice_message << " <a href=\"/lettings-logs/update-logs\">Update #{unresolved_logs_count_for_user} more logs</a>" if unresolved_logs_count_for_user.positive?
+      flash[:notice] = notice_message.html_safe
+    end
```

Comment:
```suggestion
  def resolve_logs!
    if @log&.unresolved && @log.location.present? && @log.scheme.present? && @log.resolve!
      notice_message = "You’ve updated all the fields affected by the scheme change."
      unresolved_logs_count_for_user = current_user.lettings_logs.where(unresolved: true, created_by: current_user).count
      notice_message << " <a href=\"/lettings-logs/update-logs\">Update #{unresolved_logs_count_for_user} more logs</a>" if unresolved_logs_count_for_user.positive?
      flash[:notice] = notice_message.html_safe
    end
```

The notice generation here probably doesn't want to be in the controller.

Reply:
```ruby
  def resolve_logs!
    if @log&.unresolved && @log.location.present? && @log.scheme.present? && @log.resolve!
      count = current_user.lettings_logs.where(unresolved: true, created_by: current_user).count
      flash.now[:notice] = flash_notice_for_resolved_logs(count)
    end
  end
```